src/test/java/org/logstash/beats/BeatsParserTest.java

```diff
@@ -0,0 +1,103 @@
+package org.logstash.beats;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.embedded.EmbeddedChannel;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+
+public class BeatsParserTest {
+    private Batch batch;
+    private Message message1;
+    private Message message2;
+    private final int numberOfMessage = 20;
+
+    @Before
+    public void setup() {
+        this.batch = new Batch();
+        this.batch.setProtocol(Protocol.VERSION_2);
+
+        Map map1 = new HashMap<String, String>();
+        map1.put("line", "My super event log");
+        map1.put("from", "beats");
+
+        this.message1 = new Message(1, map1);
+        this.batch.addMessage(this.message1);
+
+        for(int i = 0; i < numberOfMessage; i++) {
+            Map map2 = new HashMap<String, String>();
+            map2.put("line", "Another world");
+            map2.put("from", "Little big Adventure");
+
+            this.message2 = new Message(i + 1, map2);
```

Comment:
why do you loop, but always overwrite message2? intended? if so, feels hard to read :-)